jetty-util/src/test/java/org/eclipse/jetty/util/resource/ResourceAliasTest.java

```diff
@@ -18,93 +18,153 @@
-        assertNotNull(resource.getAlias());
-        resource = Resource.newResource(file0.getAbsoluteFile());
-        assertTrue(resource.exists());
-        assertNotNull(resource.getAlias());
-        resource = Resource.newResource(file0.toURI());
-        assertTrue(resource.exists());
-        assertNotNull(resource.getAlias());
-        resource = Resource.newResource(file0.toURI().toString());
-        assertTrue(resource.exists());
-        assertNotNull(resource.getAlias());
+        Path baseDir = workDir.getEmptyPathDir();
+
+        Path file = baseDir.resolve("test.txt");
+        FS.touch(file);
 
         try
         {
-            resource = dir.addPath("test.txt\0");
+            Path file0 = baseDir.resolve("test.txt\0");
+            if (!Files.exists(file0))
+                return;  // this file system does not suffer this problem
+
+            assertThat(file0 + " exists", Files.exists(file0), is(true));  // This is an alias!
+
+            Resource dir = Resource.newResource(baseDir);
+
+            // Test not alias paths
+            Resource resource = Resource.newResource(file);
+            assertTrue(resource.exists());
+            assertNull(resource.getAlias());
+            resource = Resource.newResource(file.toAbsolutePath());
+            assertTrue(resource.exists());
+            assertNull(resource.getAlias());
+            resource = Resource.newResource(file.toUri());
+            assertTrue(resource.exists());
+            assertNull(resource.getAlias());
+            resource = Resource.newResource(file.toUri().toString());
+            assertTrue(resource.exists());
+            assertNull(resource.getAlias());
+            resource = dir.addPath("test.txt");
+            assertTrue(resource.exists());
+            assertNull(resource.getAlias());
+
+            // Test alias paths
+            resource = Resource.newResource(file0);
             assertTrue(resource.exists());
             assertNotNull(resource.getAlias());
+            resource = Resource.newResource(file0.toAbsolutePath());
+            assertTrue(resource.exists());
+            assertNotNull(resource.getAlias());
+            resource = Resource.newResource(file0.toUri());
+            assertTrue(resource.exists());
+            assertNotNull(resource.getAlias());
+            resource = Resource.newResource(file0.toUri().toString());
+            assertTrue(resource.exists());
+            assertNotNull(resource.getAlias());
+
+            try
+            {
+                resource = dir.addPath("test.txt\0");
+                assertTrue(resource.exists());
+                assertNotNull(resource.getAlias());
+            }
+            catch (MalformedURLException e)
+            {
+                assertTrue(true);
+            }
         }
-        catch (MalformedURLException e)
+        catch (InvalidPathException e)
         {
-            assertTrue(true);
+            // this file system does not suffer this problem
```

Comment:
What does that comment mean? Does it mean that this exception will never be thrown?

Reply:
@joakime ?

Reply:
The original code ...

``` java
        File file0 = new File(__dir, "test.txt\0");	
        if (!file0.exists())	
            return;  // this file system does not suffer this problem
```

The updated code ...

``` java
try
{
    Path file0 = baseDir.resolve("test.txt\0");
} 
catch (InvalidPathException e) 
{
    // this file system does not suffer this problem
}
```

Reply:
The entire test is moot if the filesystem it's running on catches the issue with files ending in null.
That's a test from https://bugs.eclipse.org/437800